### ISSUE TITLE
Do not report DL3042 if cache mount is used

### ISSUE THREAD

**Issue by bvolkmer** [hadolint/hadolint#983]
- [ ] This is a bug report
- [x] This is a feature request
- [ ] I searched existing issues before opening this one

### Feature request

Do not report DL3042 if pip is called like:
```
RUN --mount=type=cache,target=/root/.cache/pip \
    pip install ...
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.1
```

**Comment by mikesir87** [comment]
Similar issue for DL3060 when cache used for yarn

**Comment by UnknownPlatypus** [comment]
I believe this could be closed as a duplicate of #497
